packages/core/src/channels/stream-helpers.test.ts

```
@@ -0,0 +1,221 @@
+ * Tests for packages/core/src/channels/stream-helpers.ts
+ *
+ * `extractErrorMessage`, `chunkToFallbackMessage`, and `renderBuiltInToolEvent`
+ * are pure functions — no I/O, no async behaviour, no mocking required.
+ * `renderBuiltInToolEvent` delegates to the already-tested `formatTool*`
+ * helpers in `./formatting`, so these tests verify correct routing rather
+ * than re-testing the formatting output itself.
+ */
+import { describe, expect, it } from 'vitest';
+
+import type { ToolDisplayEvent } from './types';
+import { chunkToFallbackMessage, extractErrorMessage, renderBuiltInToolEvent } from './stream-helpers';
```

ESLint

[error] `./stream-helpers` import should occur before type import of `./types`

(import/order)